lib/ts/controllers/s-expandable-control.ts

```diff
@@ -34,92 +32,109 @@
                 this.isCheckable = true;
-                this.isRadio = this.element.type === "radio";
+                this.isRadio = (<HTMLInputElement>this.element).type === "radio";
             } else {
                 this.isCollapsed = this._isCollapsedForClickable;
                 this.events = ["click", "keydown"];
             }
             this.listener = this.listener.bind(this);
-        },
+        };
+
 
         // for non-checkable elements, the initial source of truth is the collapsed/expanded
         // state of the controlled element (unless the element doesn't exist)
-        _isCollapsedForClickable: function () {
+        _isCollapsedForClickable() {
             var cc = this.controlledCollapsible;
             return cc ? !cc.classList.contains("is-expanded") : this.element.getAttribute("aria-expanded") === "false";
-        },
+        };
 
         // for checkable elements, the initial source of truth is the checked state
-        _isCollapsedForCheckable: function () {
-            return !this.element.checked;
-        },
+        _isCollapsedForCheckable() {
+            return !(<HTMLInputElement>this.element).checked;
+        };
+
 
         get controlledCollapsible() {
-            return document.getElementById(this.element.getAttribute("aria-controls"));
-        },
+            const attr = this.element.getAttribute("aria-controls");
+            if (!attr) {
+                throw "couldn't find controls"
+            }
+            const result = document.getElementById(attr);
+            if (!result) {
+                throw "couldn't find controls"
+            }
+            return result;
+        };
 
-        _dispatchShowHideEvent: function(isShow) {
+        _dispatchShowHideEvent(isShow: boolean) {
             this.triggerEvent(isShow ? "show" : "hide");
-        },
+        };
 
-        _toggleClass: function(doAdd) {
+        _toggleClass(doAdd: boolean) {
             if (!this.data.has("toggle-class")) {
                 return;
             }
             var cl = this.element.classList;
-            this.data.get("toggle-class").split(/\s+/).forEach(function (cls) {
+            var toggleClass = this.data.get("toggle-class");
+            if (!toggleClass) {
+                throw "couldn't find toggle class"
+            }
+            toggleClass.split(/\s+/).forEach(function (cls) {
                 cl.toggle(cls, !!doAdd);
             });
-        },
+        };
 
-        listener: function(e) {
+        listener(e: UIEvent) {
             var newCollapsed;
             if (this.isCheckable) {
-                newCollapsed = !this.element.checked;
+                newCollapsed = !(<HTMLInputElement>this.element).checked;
             } else {
-                if (e.type == "keydown" && (e.keyCode != 13 && e.keyCode != 32)) {
+                if (e.type == "keydown" && ((<KeyboardEvent>e).keyCode != 13 && (<KeyboardEvent>e).keyCode != 32)) {
```

Comment:
Could use a type guard here?

Reply:
Added `e instanceof KeyboardEvent &&` here as a type guard (as defined by the [TS docs](https://www.typescriptlang.org/docs/handbook/advanced-types.html#instanceof-type-guards)). Let me know if this was not what you were intending.